hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java

```diff
@@ -1872,6 +1872,12 @@ private int getWhenNoSSLDetected() {
     public static final HazelcastProperty PERSISTENCE_AUTO_CLUSTER_STATE_STRATEGY = new HazelcastProperty(
             "hazelcast.persistence.auto.cluster.state.strategy", ClusterState.NO_MIGRATION);
 
+    /**
+     * The temporary directory path to be used for uploaded jars
+     */
+    public static final HazelcastProperty JAR_UPLOAD_TEMP_DIR_PATH
```

Comment:
Isn't the name containing part `TEMP_DIR` actually confusing? IIUC it's not temp directory, it's just common directory (I understand it replaces default location which is tmp dir, but this dir isn't temp). WDYT?

Reply:
The jar is stored in the given directory and then the jar gets deleted. That is why it sounds temporary to me. I also wanted to indicate that it replaces the default temp directory of the OS
But if it is confusing I can rename it. Is JAR_UPLOAD_DIR_PATH better for example ?

Reply:
`JAR_UPLOAD_DIR_PATH` sounds better for me. IIUC the files are removed only in case they are processed correctly, right? For example if server crash then they will not never removed.

However, I don't have strong opinion on that, @frant-hartm what's your opinion about it?